### ISSUE TITLE
The tags in the select query result displays incorrect data

### ISSUE THREAD

**Issue by anand-vardham** [influxdata/influxdb#6279]
The Ids in the select query is not displayed correctly. Its displaying incorrect value for cluster_id, svm_id and the id tag. which result in the failure of the UDF script
with missing key/value error.

`SELECT iops, latency, throughput, totalcapacity, usedcapacity FROM janus16."5mins".volume_data WHERE time > '2016-04-11T09:51:46.810561587Z' AND time < '2016-04-11T10:51:46.810561587Z' and cluster_id='avardham'   GROUP BY svm_id, cluster_id, id`

```
name: volume_data
tags: cluster_id=id, id=, svm_id=avardham
time                    iops                    latency                 throughput              totalcapacity   usedcapacity
----                    ----                    -------                 ----------              -------------   ------------
1460370744000000000     0.038461538461538464    73311                   0                       102400          5300
1460371244000000000     0                       0                       0                       102400          5300
1460371331000000000     0.034482758620689655    73073.33333333333       0                       102400          5300
1460371419000000000     0.03409090909090909     74895.33333333333       0                       102400          5300
1460371524000000000     0.02857142857142857     73372                   0                       102400          5300
```

The show tag values displays the data properly

```
> show tag values with key=svm_id where cluster_id = 'avardham'
name: volume_data
-----------------
key     value
svm_id  Systemic21
svm_id  Systemic22
svm_id  Systemic23
svm_id  Systemic24
svm_id  Systemic25
svm_id  avardham-01
```

**Comment by anand-vardham** [comment]
Influxdb version is 0.11
Kapacitor version is 0.11

**Comment by jsternberg** [comment]
Can you give me a reproducer so I can try it out? Can you also try using 0.12.1 and see if the problem still exists?

**Comment by jsternberg** [comment]
Also an example of what you expect the output to be would also be very helpful.

**Comment by anand-vardham** [comment]
Hi, Thanks for replying, actually i am new to influx so could you let me know how to create the reproducer. Also i expect the output of the query to have correct values for the tags, which is as mentioned below.

name: volume_data
tags: cluster_id=avardham, id=volume1, svm_id=Systemic21
time                    iops                    latency                 throughput              totalcapacity   usedcapacity
----                    ----                    -------                 ----------              -------------   ------------
1460370744000000000     0.038461538461538464    73311                   0                       102400          5300
1460371244000000000     0                       0                       0                       102400          5300
1460371331000000000     0.034482758620689655    73073.33333333333       0                       102400          5300
1460371419000000000     0.03409090909090909     74895.33333333333       0                       102400          5300
1460371524000000000     0.02857142857142857     73372                   0                       102400          5300

So that when the same query is used inside the kapacitor as a batch tick script i do not get a missing key or value error. This error code since the id tag is set to empty.

**Comment by jsternberg** [comment]
To make a reproducer, start a completely new instance of the InfluxDB server and then open the CLI with `influx`. Here's an example:

```
$ influx
Visit https://enterprise.influxdata.com to register for updates, InfluxDB server management, and monitoring.
Connected to http://localhost:8086 version unknown
InfluxDB shell 0.9
> create database mydb
> use mydb
Using database mydb
> insert cpu,host=server01,region=uswest value=3
> select value from cpu group by host, region
name: cpu
tags: host=server01, region=uswest
time                    value
----                    -----
1460413028198208505     3
```

I tried to do something similar to what I thought you were doing but couldn't reproduce it. If you can give me the steps with the exact output that I can do on my own machine, then it'll give me a lot more information to see if there's a bug or not.

**Comment by anand-vardham** [comment]
I am able to push the data to a table using the insert statement and then view the data using select query. But when i push the data using the node js client into the influx db and then run the following query. It produces the results mentioned above.

SELECT * FROM janus16."5mins".volume_data WHERE time > now() - 1h and cluster_id = 'avardham' and svm_id='Systemic21' group by id,svm_id,cluster_id.

But if i remove the criteria 'time > now() - 1h' from the query the results are displayed properly.

**Comment by jsternberg** [comment]
@anand-vardham in order to help you, I need to see code or have the ability to redo what you did so I can see it on my computer. You can copy and paste the output from your terminal using the command line and then I can see more about what you did.

Please also surround code with code blocks so it's easier to read. I've modified the above comment to include the code blocks. For more details about markdown, you can click on the [Styling with Markdown is supported](https://guides.github.com/features/mastering-markdown/) button on the bottom left of the text window.

**Comment by katirena** [comment]
I've noticed the same issue with version 0.11.1-1 in a cluster. Sometimes happens, sometimes doesnt:

```
> create database test3
> use test3
Using database test3
> INSERT something,location=home,day=monday value=0
> select value from something group by location, day
name: something
tags: day=location, location=                           <--- here
time                    value
----                    -----
1460492513484640778     0
```


```
> create database test4
> use test4
Using database test4
> INSERT something,location=home,day=monday value=0
> select value from something group by location, day
name: something
tags: day=monday, location=home
time                    value
----                    -----
1460492601249029355     0
```

```
> create database test5
> use test5
Using database test5
> INSERT something,location=home,day=monday value=1
> select value from something group by location, day
name: something
tags: day=location, location=                            <--- and here
time                    value
----                    -----
1460492808095557956     1

```

**Comment by jsternberg** [comment]
I haven't confirmed it, but #6190 might be relevant.

Is it possible to see if you can reproduce with 0.12.1? That change is in there. Unfortunately you mentioned clustering which was removed from 0.12. You might be able to cherry pick that commit to the 0.11 branch and try building a custom version with that.

**Comment by jsternberg** [comment]
@jwilder do you think this is related to #6190?

**Comment by jwilder** [comment]
Don't think so. Sounds like it might be related to distributed queries in 0.11.

**Comment by jsternberg** [comment]
We finally found the reason for this and fixed it in fb7be8c. We no longer support 0.11 which would be the only open source version to have this problem because it's the last one to support clustering, but you can still take this commit and backport it to 0.11 if you want to continue trying to use clustering in 0.11.

Thank you for the bug report. I'm going to close this since there's nothing else for us to do on this issue.
